File: sizzle/lib/entities/productmin.dart
class ProductMin {
  final int id;
  final String name;
  final String imageUrl;
  final double rating;
  final int price;
  final int ratingsCount;
  final double discountPercentage;

  const ProductMin({
    required this.id,
    required this.name,
    required this.imageUrl,
    required this.rating,
    required this.price,
    required this.ratingsCount,
    required this.discountPercentage,
  });

  factory ProductMin.fromJson(Map<String, dynamic> json) {
    return ProductMin(
      id: json['id'] as int,
      name: json['name'] as String,
      imageUrl: json['imageUrl'] as String,
      rating: (json['rating'] as num).toDouble(),
      price: json['price'] as int,
      ratingsCount: json['ratingsCount'] as int,
      discountPercentage: (json['discountPercentage'] as num).toDouble(),
    );
  }
}

File: backend/sizzle-productlist/src/main/java/com/sizzle/products/sizzle_productlist/dto/ProductMin.java
package com.sizzle.products.sizzle_productlist.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
public class ProductMin {
    final Long id;
    final String name;
    final String imageUrl;
    final float rating;
    final int price;
    final int ratingsCount;
    final float discountPercentage;

    public ProductMin(Long id,String name, String imageUrl,float rating, int price, int ratingsCount, float discountPercentage){
        this.id = id;
        this.name=name;
        this.imageUrl=imageUrl;
        this.rating=rating;
        this.price=price;
        this.ratingsCount=ratingsCount;
        this.discountPercentage=discountPercentage;
    }
}
